discovery_engine/lib/src/domain/search_manager.dart

```diff
@@ -121,23 +123,26 @@ class SearchManager {
     );
     final docs = await _getSearchDocuments(search);
     await _searchRepo.save(search);
-    return EngineEvent.searchRequestSucceeded(search, docs);
+    return EngineEvent.searchRequestSucceeded(search.toApiRepr(), docs);
   }
 
   /// Obtain the next batch of search documents and persist to repositories.
   Future<EngineEvent> nextSearchBatchRequested() async {
-    var search = await _searchRepo.getCurrent();
+    final search = await _searchRepo.getCurrent();
 
     if (search == null) {
       const reason = SearchFailureReason.noActiveSearch;
       return const EngineEvent.nextSearchBatchRequestFailed(reason);
     }
 
     // lets update active search params
-    search = search.copyWith(requestedPageNb: search.requestedPageNb + 1);
+    search.requestedPageNb += 1;
```

Comment:
🔧 if we change the field directly, then we could remove the unused `ActiveSearch.nextPage()` method completely. is there a specific reason why the method is not used anymore (independent of whether the method mutates the object or creates a new object)?

❓ we probably can't mark the fields of `ActiveSearch` as private because of `hive`, right? otherwise we could better contain the changes to the object.

Reply:
> we probably can't mark the fields [...] private [...]

Also we use the fields in various places outside of the module so we would need to add getters and and the `nextPage` method again which conflicts with the feedback from @acrrd :

> Also this data is a basically a POD,  it should not contain any logic about what the next page is.

It's also not in line with the other types in `(domain|api)/models/` which are all PODs.

Reply:
>  [..] could remove [..]

done